pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -154,6 +157,13 @@ type PriorityQueue struct {
 
 	cond sync.Cond
 
+	// inFlightPods holds the UID of all pods which have been popped out for which Done
+	// hasn't been called yet - in other words, all pods that are currently being
+	// processed.
+	inFlightPods map[types.UID]inFlightPod
+	// receivedEvents holds the events received by the scheduling queue.
+	receivedEvents []clusterEvent
```

Comment:
Can we have an Error log line for cases when these lists grow too much?

If the scheduler crashes or gets high latency, we would like to know whether it's because of this.
Is there a metric we can use to monitor?

Reply:
I guess the goroutines metric is a proxy for `inFlightPods`, but not for `receivedEvents`

Reply:
> Is there a metric we can use to monitor?

We don't have it now, I think. If we want to have an error log, which should be the threshold to log it? (I mean how many receivedEvents are when we tell users about the huge list as an error?) Maybe it depends on how much memory available for the scheduler, right?
So, is it better to have a new metric to observe the number of receivedEvents, instead of having some error logs about it?

Reply:
cc @ahg-g for feedback

Reply:
@ahg-g Any thought?